src/Blumchen.DependencyInjection/Workers/Worker.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Concurrent;
+using System.Text.Json.Serialization;
+using Blumchen.Configuration;
+using Blumchen.Serialization;
+using Blumchen.Subscriptions;
+using Blumchen.Subscriptions.Management;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+
+namespace Blumchen.Workers;
+
+public abstract class Worker<T>(
+    DatabaseOptions databaseOptions,
+    IHandler<T> handler,
+    JsonSerializerContext jsonSerializerContext,
+    IErrorProcessor errorProcessor,
+    ResiliencePipeline pipeline,
+    INamingPolicy namingPolicy,
+    PublicationManagement.PublicationSetupOptions publicationSetupOptions,
+    ReplicationSlotManagement.ReplicationSlotSetupOptions replicationSlotSetupOptions,
+    Func<TableDescriptorBuilder,TableDescriptorBuilder> tableDescriptorBuilder,
+    ILoggerFactory loggerFactory): BackgroundService where T : class
+{
+    private readonly ILogger<Worker<T>> _logger = loggerFactory.CreateLogger<Worker<T>>();
```

Comment:
**CONCERN:** Any reason why not to just inject ILogger<Worker<T>>?

Reply:
ATM we have to pass it to [NpgsqlDataSourceBuilder](https://github.com/event-driven-io/Blumchen/blob/master/src/Blumchen/Subscriptions/Subscription.cs#L41). We may review it in a (near) future release

Reply:
Done